### DISCUSSION TITLE
Change default service.name of otel collector

### DISCUSSION THREAD

**Discussion by jan-machacek-kosik** [open-telemetry/opentelemetry-collector#9898] [Q&A]
Is there a way to change the default service.name or service.environment from 'otelcol-contrib' to something else?

I tried to set service.name using the OTEL_SERVICE_NAME environment variable, but it didn't work.

**Comment by aloisiobilck** [comment]
I have the same problem

**Comment by LiamDavey** [comment]
Doesn't automatically use OTEL_SERVICE_NAME, but can be configured that way if you set the attribute yourself following [this](https://opentelemetry.io/docs/collector/internal-telemetry/#configure-resource-attributes), and setting it to an [environment variable](https://opentelemetry.io/docs/collector/configuration/#environment-variables) like below.
```
service:
  telemetry:
    resource:
      service.name: ${env:OTEL_SERVICE_NAME}
```
